packages/core/src/fact-evaluator.ts

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * This file provides utilities for creating fact evaluation functions
+ * that can be used with the test evaluation framework.
+ * 
+ * The actual implementation that uses the classify runtime helper
+ * should be provided by higher-level packages that have access to both
+ * core and runtime dependencies.
+ */
+
+import type { FactEvaluationFunction } from "./testeval.js";
+
+/**
+ * Creates a fact evaluation function using a classify function.
+ * This is a factory function that allows packages with access to the
+ * classify runtime helper to create compatible fact evaluation functions.
+ * 
+ * @param classifyFn - The classify function from the runtime package
+ * @returns A fact evaluation function compatible with test evaluation
+ */
+export function createClassifyBasedFactEvaluator(
+  classifyFn: (
+    text: string,
+    labels: Record<string, string>,
+    options?: any
+  ) => Promise<{ label: string; answer: string; error?: string }>
+): FactEvaluationFunction {
+  return async (outputText: string, fact: string) => {
+    try {
+      // Use classify to determine if output is factually consistent with the fact
+      const result = await classifyFn(
+        `Output: ${outputText}\n\nFact to verify: ${fact}`,
```

Comment:
use the overload that provides a chatgeneation context and use `def` to store the outputText and fact.